### PR TITLE
Plugin is now stoppable via #stop

### LINKED ISSUE #10: refactor shutdown sequence to use plugin.stop
Change shutdown sequence to be triggered by plugin.stop instead of ShutdownSignal exception.
Also remove any calls to: shutdown, finished, finished?, running? or terminating?
This depends on https://github.com/elastic/logstash/pull/3895

### PR DESCRIPTION
**Pull request by andrewvc**
This fixes #10

### CODE REVIEW

File: lib/logstash/inputs/rss.rb [hunk 1]
```diff
@@ -25,57 +28,20 @@ class LogStash::Inputs::Rss < LogStash::Inputs::Base
...
   def register
-    require "faraday"
-    require "rss"
     @logger.info("Registering RSS Input", :url => @url, :interval => @interval)
   end # def register
 
   public
   def run(queue)
+    @run_thread = Thread.current
     loop do
```
Comment (jordansissel) [inline lib/logstash/inputs/rss.rb]: Myabe instead of 'break if stop?' we can have `while !stop?` for the loop?

Comment (andrewvc) [inline lib/logstash/inputs/rss.rb]: will fix

**Comment by jordansissel** [comment]
photo on 9-22-15 at 3 31 pm

**Comment by jordansissel** [comment]
LGTM also

**Pull request closed without merging**